powerauth-nextstep-model/src/main/java/io/getlime/security/powerauth/lib/nextstep/model/entity/attribute/OperationFormAttribute.java

```diff
@@ -0,0 +1,99 @@
+package io.getlime.security.powerauth.lib.nextstep.model.entity.attribute;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import java.util.Objects;
+
+/**
+ * Abstract class that represents an attribute used during the operation.
+ *
+ * This class requires JSON annotation since it is a base class that is extended with specific types.
+ *
+ * @author Petr Dvorak, petr@lime-company.eu
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = OperationTitleAttribute.class, name = "TITLE"),
+        @JsonSubTypes.Type(value = OperationMessageAttribute.class, name = "MAIN_MESSAGE"),
+        @JsonSubTypes.Type(value = OperationAmountAttribute.class, name = "AMOUNT"),
+        @JsonSubTypes.Type(value = OperationNoteAttribute.class, name = "MESSAGE"),
+        @JsonSubTypes.Type(value = OperationBankAccountChoiceAttribute.class, name = "BANK_ACCOUNT_CHOICE"),
+        @JsonSubTypes.Type(value = OperationKeyValueAttribute.class, name = "KEY_VALUE")
+})
+public class OperationFormAttribute {
+
+    // TODO - revise types MAIN_MESSAGE AND MESSAGE to avoid confusion
+    public enum Type {
+        TITLE,
+        MAIN_MESSAGE,
+        AMOUNT,
+        MESSAGE,
+        BANK_ACCOUNT_CHOICE,
+        KEY_VALUE
+    }
+
+    // JsonIgnore added, otherwise type was serialized twice
+    @JsonIgnore
+    protected Type type;
+
+    protected String id;
+
+    protected String label;
```

Comment:
"label" does not feel like generic attribute key, it is related to the particular display of the attribute of given type... "type" and "id" should be enough here.

Reply:
Introduction of generic label made the internationalization much easier - I can just call attribute.setLabel(localize(id)) but I can put it back into individual subclasses, it doesn't matter so much.

Reply:
OK, it makes sense. Thank you for explanation.